src/main/java/org/folio/services/ResponseHandlerUtil.java

```diff
@@ -0,0 +1,60 @@
+package org.folio.services;
+
+import static javax.ws.rs.core.HttpHeaders.CONTENT_TYPE;
+
+import javax.ws.rs.core.Response;
+import org.folio.rest.jaxrs.model.Errors;
+
+public final class ResponseHandlerUtil {
+  private static final String HRID_ERROR_MESSAGE = "lower(f_unaccent(jsonb ->> 'hrid'::text))";
+  private static final String HRID = "HRID";
+
+  private ResponseHandlerUtil() {
+  }
+
+  public static Response handleHridError(Response response) {
+    var statusCode = response.getStatus();
+
+    if (statusCode == 201) {
+      return response;
+    }
+
+    var errorMessage = getErrorMessage(response.getEntity());
+    if (errorMessage.contains(HRID_ERROR_MESSAGE)
+      && errorMessage.contains("instance") && statusCode == 400) {
+      return createResponse(400, errorMessage);
+    } else if (errorMessage.contains(HRID_ERROR_MESSAGE)
+      && (errorMessage.contains("item") || errorMessage.contains("holdings_record")) && statusCode == 422) {
+      return createResponse(422, errorMessage);
```

Comment:
You could pass original Response object, to avoid duplicate/pass statuses and additional conditions for content type.
It'd be good also to keep original object for 422 (with error parameters) and replace only desired message